src/cli/init.js

```diff
@@ -0,0 +1,158 @@
+// @flow
+// Implementation of `sourcecred init`
+
+import dedent from "../util/dedent";
+import {type RepoId, stringToRepoId} from "../core/repoId";
+import {type Project, projectToJSON} from "../core/project";
+import type {Command} from "./command";
+import * as Common from "./common";
+import fs from "fs-extra";
+import process from "process";
+import read from "read";
+import path from "path";
+import {fetchGithubOrg} from "../plugins/github/fetchGithubOrg";
+import {type DiscourseServer} from "../plugins/discourse/loadDiscourse";
+
+function usage(print: (string) => void): void {
+  print(
+    dedent`\
+    usage: sourcecred init
+           sourcecred init --help
+
+    Sets up a new SourceCred instance, by interactively creating a SourceCred
+    project configuration, and saving it to 'sourcecred.json'.
+
+    Arguments:
+        --help
+            Show this help message and exit, as 'sourcecred help init'.
+
+    Environment variables:
+        SOURCECRED_GITHUB_TOKEN
+            API token for GitHub. This should be a 40-character hex
+            string. Required if you want to load whole GitHub orgs.
+
+            To generate a token, create a "Personal access token" at
+            <https://github.com/settings/tokens>. When loading data for
+            public repositories, no special permissions are required.
+            For private repositories, the 'repo' scope is required.
+    `.trimRight()
+  );
+}
+
+function die(std, message) {
+  std.err("fatal: " + message);
+  std.err("fatal: run 'sourcecred help init' for help");
+  return 1;
+}
+
+const initCommand: Command = async (args, std) => {
+  for (let i = 0; i < args.length; i++) {
+    switch (args[i]) {
+      case "--help": {
+        usage(std.out);
+        return 0;
+      }
+      default: {
+        return die(std, `Unexpected argument ${args[i]}`);
+      }
+    }
+  }
+  const dir = process.cwd();
+  const projectFilePath = path.join(dir, "sourcecred.json");
+  if (await fs.exists(projectFilePath)) {
+    return die(std, `Refusing to overwrite sourcecred.json file in ${dir}`);
```

Comment:
I think this sums up the cautionary tale of interactive scripts well.

For the majority of settings I'm inclined to use a config format which supports comments, set up the minimum with interaction, and rely on easy to grock, well documented config and examples.

In this case the absolute minimum would be the project name.

Though the utility of the most common settings I can also see. Particularly scanning an organization to add individual repo names. That might also be a good case to make this a separate utility we can use for editing our config as well. An "automatically add missing repositories" could be really handy for large multi-repo orgs.

Not blocking, but wonder what your thoughts are on this.

Reply:
I think you're right, the interactive approach is really cumbersome. I've generated a new PR at #1515 which is a non-interactive approach, kind of based on the gen-project command. Please take a look over there; I'm going to close this PR as obsolete.